src/streaming/ManifestLoader.js

```diff
@@ -338,57 +251,28 @@ function ManifestLoader() {
-                    // overwrite the manifest's duration attribute. This is a patch for if a manifest is generated incorrectly.
-                    if (settings &&
-                        settings.get().streaming.enableManifestDurationMismatchFix &&
-                        manifest.mediaPresentationDuration &&
-                        manifest.Period.length > 1) {
-                        const sumPeriodDurations = manifest.Period.reduce((totalDuration, period) => totalDuration + period.duration, 0);
-                        if (!isNaN(sumPeriodDurations) && manifest.mediaPresentationDuration > sumPeriodDurations) {
-                            logger.warn('Media presentation duration greater than duration of all periods. Setting duration to total period duration');
-                            manifest.mediaPresentationDuration = sumPeriodDurations;
-                        }
+                },
+                error: function (request, statusText, errorText) {
+                    if (!linkPeriod) {
+                        eventBus.trigger(Events.INTERNAL_MANIFEST_LOADED, {
+                            manifest: null,
+                            error: new DashJSError(
+                                Errors.MANIFEST_LOADER_LOADING_FAILURE_ERROR_CODE,
+                                Errors.MANIFEST_LOADER_LOADING_FAILURE_ERROR_MESSAGE + `${url}, ${errorText}`
+                            )
+                        });
+                    } else {
+                        reject();
                     }
-
-                    manifest.baseUri = baseUri;
-                    manifest.loadedTime = new Date();
-                    xlinkController.resolveManifestOnLoad(manifest);
-                    eventBus.trigger(Events.ORIGINAL_MANIFEST_LOADED, { originalManifest: data });
-                } else {
-                    eventBus.trigger(Events.INTERNAL_MANIFEST_LOADED, {
-                        manifest: null,
-                        error: new DashJSError(
-                            Errors.MANIFEST_LOADER_PARSING_FAILURE_ERROR_CODE,
-                            Errors.MANIFEST_LOADER_PARSING_FAILURE_ERROR_MESSAGE + `${url}`
-                        )
-                    });
-                }
-            },
-            error: function (request, statusText, errorText) {
-                if (!linkPeriod) {
-                    eventBus.trigger(Events.INTERNAL_MANIFEST_LOADED, {
-                        manifest: null,
-                        error: new DashJSError(
-                            Errors.MANIFEST_LOADER_LOADING_FAILURE_ERROR_CODE,
-                            Errors.MANIFEST_LOADER_LOADING_FAILURE_ERROR_MESSAGE + `${url}, ${errorText}`
-                        )
-                    });
                 }
-            }
-        });
+            });
+        }
+
+        if (linkPeriod) {
+            return new Promise(loadUrl);
+        } else {
+            loadUrl();
+        }
```

Comment:
Suggestion: we can move this logic to the `if conditional` in line 134 and reuse it